pkg/apis/kf/v1alpha1/app_defaults_test.go

```diff
@@ -30,3 +32,58 @@ func TestAppSpec_SetDefaults_BlankContainer(t *testing.T) {
 	testutil.AssertEqual(t, "len(spec.template.spec.containers)", 1, len(app.Spec.Template.Spec.Containers))
 	testutil.AssertEqual(t, "spec.template.spec.containers.name", "", app.Spec.Template.Spec.Containers[0].Name)
 }
+
+func TestAppSpec_SetDefaults_ResourceLimits_Default(t *testing.T) {
+	t.Parallel()
+
+	app := &App{
+		Spec: AppSpec{
+			Template: AppSpecTemplate{
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{{}},
+				},
+			},
+		},
+	}
+	app.SetDefaults(context.Background())
+
+	wantMem := resource.MustParse("1Gi")
+	wantStorage := resource.MustParse("1Gi")
+	wantCPU := resource.MustParse("1")
+
+	testutil.AssertEqual(t, "default memory request", wantMem, app.Spec.Template.Spec.Containers[0].Resources.Requests[v1.ResourceMemory])
```

Comment:
maybe extract `app.Spec.Template.Spec.Containers[0].Resources.Requests` into a variable rather than access it each time